src/cwt.rs

```diff
@@ -0,0 +1,135 @@
+use serde::{Deserialize, Serialize};
+use serde_cbor::Value;
+use std::collections::BTreeMap;
+use std::ops::Deref;
+
+pub use crate::claim::{self, Claim, Label, NumericDate};
+
+/// CWT claims set.
+#[derive(Clone, Debug, Default, Deserialize, Serialize, PartialEq, Eq)]
+pub struct ClaimsSet(BTreeMap<Value, Value>);
+
+impl ClaimsSet {
+    pub fn insert_claim<T: Claim>(&mut self, claim: T) -> Option<Value> {
+        self.0.insert(T::label().into(), claim.into())
+    }
+
+    pub fn get_claim<T: Claim>(&self) -> Option<&Value> {
+        self.0.get(&T::label().into())
+    }
+
+    pub fn serialize(&self) -> Result<Vec<u8>, Error> {
+        serde_cbor::to_vec(&self).map_err(Error::UnableToSerializeClaimsSet)
+    }
+}
+
+impl AsRef<BTreeMap<Value, Value>> for ClaimsSet {
+    fn as_ref(&self) -> &BTreeMap<Value, Value> {
+        &self.0
+    }
+}
+
+impl Deref for ClaimsSet {
+    type Target = BTreeMap<Value, Value>;
+
+    fn deref(&self) -> &BTreeMap<Value, Value> {
+        &self.0
+    }
+}
+
+impl From<BTreeMap<Value, Value>> for ClaimsSet {
+    fn from(m: BTreeMap<Value, Value>) -> Self {
+        Self(m)
+    }
+}
+
+/// Errors that can occur working with a ClaimsSet.
+#[derive(Debug, thiserror::Error)]
+pub enum Error {
+    #[error("unable to serialize CWT ClaimsSet: {0}")]
+    UnableToSerializeClaimsSet(serde_cbor::Error),
```

Comment:
follows conventions of errors defined in `sign1.rs` -- but is it an antipattern to define errors in two separate modules? Is it preferred to re-export or organize errors somehow else?

Reply:
I think we need to put more thought into how error handling works in our libraries. Out of scope for this issue I think, this is good enough as it is the most flexible but potentially clunky solution